dev/date_binding.h

```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+#include "functional/cxx_core_features.h"
+
+// need support for #include <compare> for advanced std::chrono members
+#if defined(SQLITE_ORM_THREE_WAY_COMPARISON_SUPPORTED)
+
+#include <chrono>
+#include <sstream>
+#include <regex>
+#include <format>
+
+inline std::chrono::sys_days Today() {
+    const auto today = std::chrono::sys_days{floor<std::chrono::days>(std::chrono::system_clock::now())};
+    return today;
+}
+
+/**
+ *  This is the date we want to map to our sqlite db.
+  *  Let's make it `TEXT`
+  */
+
+//  also we need transform functions to make string from enum..
+inline std::string SysDaysToString(std::chrono::sys_days pt) {
+    auto r = std::format("{:%F}", pt);
+    return r;
+}
+
+constexpr auto null_sys_day = std::chrono::sys_days{std::chrono::days{0}};
+
+/**
+ *  and sys_days from string. This function has nullable result cause
+ *  string can be neither `male` nor `female`. i have taken null_sys_day to
+ *  correspond to null. Don't know if should use std::optional as return case??
+ *  Of course we won't allow this to happen but as developers we must have
+ *  a scenario for this case.
+ *  These functions are just helpers. They will be called from several places
+ *  that's why I placed it separatedly. You can use any transformation type/form
+ *  (for example BETTER_ENUM https://github.com/aantron/better-enums)
+ */
+inline std::chrono::sys_days SysDaysFromString(const std::string& s) {
+    using namespace std::literals;
+    using namespace std::chrono;
+
+    std::stringstream ss{s};
+    std::chrono::sys_days tt;
+    ss >> std::chrono::parse("%F"s, tt);
+    if(!ss.fail()) {
+        return tt;
+    }
+    return null_sys_day;
+}
+
+/**
+ *  This is where magic happens. To tell sqlite_orm how to act
+ *  with SysDays we have to create a few service classes
+ *  specializations (traits) in sqlite_orm namespace.
+ */
+namespace sqlite_orm {
+
+    /**
+	 *  First of all is a type_printer template class.
+	 *  It is responsible for sqlite type string representation.
+	 *  We want SysDays to be `TEXT` so let's just derive from
+	 *  text_printer. Also there are other printers: real_printer and
+	 *  integer_printer. We must use them if we want to map our type to `REAL` (double/float)
+	 *  or `INTEGER` (int/long/short etc) respectively.
+	 */
+    template<>
+    struct type_printer<std::chrono::sys_days> : public text_printer {};
```

Comment:
what if user wants to store `sys_days` as integer?

Reply:
he would have to provide his own implementation for sys_days binding...

Reply:
I chose as text because the you can see the date in a sql client... an integer would be like # of days since X for some X and it is not a meaningful representation

Reply:
but I moved from unit test to example
and I modified CMakeLists.txt in examples folder so that you can build projects to compile the examples

Reply:
> he would have to provide his own implementation for sys_days binding...

Reply:
> he would have to provide his own implementation for sys_days binding...

No one cannot define specialization which is already defined

Reply:
> I chose as text because the you can see the date in a sql client... an integer would be like # of days since X for some X and it is not a meaningful representation

Sounds like a personal preferences. I prefer keeping data as integer in unix format cause it is comfortable to make operations on it. To see the date from unix time in SQL client you only need to add one function call. One more advantage: you don't need to keep date string format when you store date as integer

Reply:
it is still here. IDK why you marked this conversation as resolved